src/main/java/com/exactpro/th2/store/common/AbstractStorage.java

```diff
@@ -18,20 +18,22 @@
 import org.slf4j.LoggerFactory;
 
 import com.exactpro.cradle.CradleManager;
-import com.exactpro.cradle.utils.CradleStorageException;
+import com.exactpro.cradle.CradleStorage;
 import com.exactpro.th2.common.schema.message.MessageRouter;
 import com.exactpro.th2.common.schema.message.SubscriberMonitor;
 
 public abstract class AbstractStorage<T> {
 
     protected final Logger logger = LoggerFactory.getLogger(this.getClass() + "@" + this.hashCode());
 
-    private final MessageRouter<T> router;
+    protected final MessageRouter<T> router;
+    protected final CradleStorage cradleStorage;
```

Comment:
Why do we need a separate variable for `CradleStorage`? We can get it at any time using the `getCradleManager` method and calling `getStorage` after that

Reply:
Yes, but we need only the CradleStorage in child implementations.